topics/reference_guide/custom_language_support/inlay_hints.md

```diff
@@ -0,0 +1,79 @@
+[//]: # (title: Inlay Hints)
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+[Inlay Hints](https://www.jetbrains.com/help/idea/viewing-reference-information.html#inlay-hints)
+render small pieces of information directly into the editor and give developers additional code insight without disturbing the workflow.
+A well-known example is parameter hints that usually display the name of the function parameters as given in its declaration.
+They are closely related to [](parameter_info.md) which shows parameter types for all possible overloads of a function but opens as a popup overlaying the code.
+
+Inlay hints are flexible and have a wide range of applications in the IntelliJ Platform.
+For instance, the following are well-known examples where inlay hints are used:
+
+- Java uses inlays to display type annotations in
+  [Java chained calls](https://www.jetbrains.com/help/idea/inlay-hints-java.html#method_chains).
+- In version-controlled projects, the author of the code is shown using inlay hints.
+- Kotlin uses inlays in range expressions to show, e.g. less-than, or less-than-or-equal signs to let
+  developers know if intervals are inclusive or exclusive.
+
+The IntelliJ Platform offers two extension points (EP) that plugin developers can implement to create inlay hints:
```

Comment:
I'm an IntelliJ plugin author and as I was reading through this docs, I was very confused (still am) on how to actually hook into these extension points from my plugin.

Is there possibly a page you could link to here, or perhaps some sample code?

Reply:
would you mind asking your question on the plugin development Slack or Forum? and please clarify what you mean with "hook into" exactly? thanks! https://plugins.jetbrains.com/docs/intellij/getting-help.html#problems-with-code-support-issues